dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/CompanyDataController.kt

```diff
@@ -47,19 +54,49 @@ class CompanyDataController(
         )
         return ResponseEntity.ok(
             companyManager.searchCompaniesAndGetApiModel(
-                CompanySearchFilter(
-                    searchString ?: "",
-                    onlyCompanyNames,
-                    dataTypes ?: setOf(),
-                    countryCodes ?: setOf(),
-                    sectors ?: setOf(),
-                    onlyCurrentUserAsUploader,
+                StoredCompanySearchFilter(
+                    searchString = searchString ?: "",
+                    nameOnlyFilter = onlyCompanyNames,
+                    dataTypeFilter = dataTypes.map { it.name },
+                    countryCodeFilter = countryCodes?.toList() ?: listOf(),
+                    sectorFilter = sectors?.toList() ?: listOf(),
+                    uploaderIdFilter = getUploaderIdFilter(onlyCurrentUserAsUploader),
                 ),
                 DatalandAuthentication.fromContextOrNull(),
             ),
         )
     }
 
+    private fun getUploaderIdFilter(onlyCurrentUserAsUploader: Boolean): List<String> {
```

Comment:
Ich erinner mich noch, dass es einen guten Grund dafür gab, aber könnt ihr ihn mir nochmal kurz erklären:

Wieso kann man mehrere uploadIds angeben beim Filter?

Reply:
Eigentlich unnötig. Ich ändere es. To be pushed.

Reply:
Frag am besten nochmal ein zwei andere aus dem Team.  Irgendwie sagt mir mein Unterbewusstsein, dass es einen guten Grund gab.  Vllt ist der aber auch Zukunftsmusik.

Reply:
👍